File: compiler/Typescript/model/node_type.ts

export enum NodeType {
	IntTypeNode = 1,
	DoubleTypeNode = 2,
	StringTypeNode = 3,
	CharTypeNode = 4,
	WhenNode = 5,
	IfNode = 6,
	ElseNode = 7,
	PackageNode = 8,
	ImportNode = 9,
	EnumNode = 10,
	StructNode = 11,
	ClassNode = 12,
	FunctionNode = 13,
	PrivateNode = 14,
	ValNode = 15,
	VarNode = 16,
	LetNode = 17,
	CharArrayNode = 18,
	UnknownKeyWordNode = 19,
	UnknownSignNodeNode = 20,

	// math symobols
	ModulusNode = 21,
	AddNode = 22,
	MinusNode = 23,
	DivideNode = 24,
	SubtractNode = 25,
	MultiplyNode = 26,

	// space Node
	SpaceNode = 27,

	// unif signs
	SquareBracketOpenNode = 28,
	SquareBracketCloseNode = 29,
	CurvedBracketOpenNode = 30,
	CurvedBracketCloseNode = 31,
	CurlyBracketOpenNode = 32,
	CurlyBracketCloseNode = 33,

	SemiColonNode = 34,
	ColonNode = 35,

	// comment
	LineComment = 36,
	MultiLineComment = 37,

	// stirn 
	QuotationNode = 38,

	//new 
	NewLine = 39

}

File: compiler/Java/src/model/NodeType.java
package model;

public enum NodeType {

    // keywords
    ValNode,
    VarNode,
    LetNode,
    ListNode,
    ClassNode,
    FunctionNode,
    StructNode,
    EnumNode,
    PackageNode,
    ImportNode,
    PrivateNode,
    ReturnNode,
    SomeNode,
    NoneNode,
    NotNodeType,
    AndNode,
    AliasNode,
    ThenNode,

    // types
    IntegerNode,
    DoubleNode,
    StringNode,
    StringValue,
    CharacterNode,
    IfNode,
    ElseNode,
    WhenNode,
    ImplementsNode,
    ExtendsNode,

    // operation node

    AndAndNode,
    OrNode,
    AndOrNode,


    // binary node

    BinaryAndNode ,
    BinaryOrNode ,




    // operators
    DivideNode,
    MultiplyNode,
    ModulasNode,
    AddNode,
    AssignNode,
    SubtractNode,
    NotEqualToNode,
    EquateNode,
    LessThan,
    LessThanOrEqualTo,
    GreaterThan,
    GreaterThanOrEqualTo,


    //    symbols
    SquareBracketOpenNode,
    SquareBracketCloseNode,
    CurvedBracketOpenNode,
    CurvedBracketCloseNode,
    CurlyBracketOpenNode,
    CurlyBracketCloseNode,
    LineCommentNode,
    MultiLineCommentNode,
    SemiColonNode,
    ColonNode,
    NewLineNode,
    SpaceCharacterNode,
    NumericNode,
    NegativeNumericNode,
    AlphaNumericNode,
    EndOfFileNode,
    UnknownNode,




}
